dask/dataframe/core.py

```diff
@@ -4951,14 +4951,19 @@ def quantile(df, q, method="default"):
 
     # currently, only Series has quantile method
     if isinstance(df, Index):
-        meta = pd.Series(df._meta_nonempty).quantile(q)
+        series_typ = type(df._meta.to_series())
+        meta = series_typ(df._meta_nonempty).quantile(q)
     else:
+        if is_series_like(df._meta):
+            series_typ = type(df._meta)
+        else:
+            series_typ = type(df._meta[df._meta.columns[0]])
```

Comment:
This will fail for duplicate columns.

```python
df = pd.DataFrame(columns=['a', 'a'])
ddf = dd.from_pandas(df, 2)
```

Reply:
```python
>>> df = pd.DataFrame(columns=['a', 'a'])
>>> ddf = dd.from_pandas(df, 2)
>>> ddf._meta
Empty DataFrame
Columns: [a, a]
Index: []
>>> ddf._meta.columns[0]
'a'
>>> ddf._meta[ddf._meta.columns[0]]
Empty DataFrame
Columns: [a, a]
Index: []
>>> type(ddf._meta[ddf._meta.columns[0]])
<class 'pandas.core.frame.DataFrame'>
>>> ddf.quantile()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/gali/Documents/GitHub/dask/dask/dataframe/core.py", line 1876, in quantile
    meta = self._meta.quantile(q, axis=axis)
  File "/Users/gali/anaconda3/lib/python3.7/site-packages/pandas/core/frame.py", line 7778, in quantile
    transposed=is_transposed)
  File "/Users/gali/anaconda3/lib/python3.7/site-packages/pandas/core/internals/managers.py", line 483, in quantile
    values = _concat._concat_compat([b.values for b in blocks])
  File "/Users/gali/anaconda3/lib/python3.7/site-packages/pandas/core/dtypes/concat.py", line 170, in _concat_compat
    return np.concatenate(to_concat, axis=axis)
ValueError: need at least one array to concatenate
>>> df.quantile()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/gali/anaconda3/lib/python3.7/site-packages/pandas/core/frame.py", line 7778, in quantile
    transposed=is_transposed)
  File "/Users/gali/anaconda3/lib/python3.7/site-packages/pandas/core/internals/managers.py", line 483, in quantile
    values = _concat._concat_compat([b.values for b in blocks])
  File "/Users/gali/anaconda3/lib/python3.7/site-packages/pandas/core/dtypes/concat.py", line 170, in _concat_compat
    return np.concatenate(to_concat, axis=axis)
ValueError: need at least one array to concatenate
```

@TomAugspurger The only motive of this is to extract the `DataFrame` type, so I suppose even if it has duplicate columns this will work as desired.